### DISCUSSION TITLE
Pick the most recent chart for ambiguous version matches

### DISCUSSION THREAD

**Discussion by wingsofovnia** [fluxcd/flux2#355] [Proposals] [answered]
Currently, `ChartRepository#Get` sorts charts as per Semver para 11 (in reverse order) before returning the top matching chart version. However, `build metadata does not figure into precedence` as per spec. This means if two versions have equal `MAJOR`, `MINOR`, `PATCH` tokens, but different build metadata (para 10), the order is 💥 **undefined**. 

Given that we don't simply match version string but **chart** versions, we have additional metadata, such as `chartVersion.Created`, that we can use to further clarify the logic without violating the Semver spec. 

Such a change is useful for dev environments that are expected to get the latest build of the same app/chart. In the current setup, this cannot be done, especially when (and usually does) build metadata contains data that doesn't sort well (git hashes, sigs, etc.).

Implementation: https://github.com/fluxcd/source-controller/pull/172

cc @hiddeco @stefanprodan

**Comment by hiddeco** [comment]
Due to the fact that we try to align features across source-controller source types, how would this work for `GitRepository` resources that make use of a semver tag strategy?

**Reply by wingsofovnia** [reply]
It is possible to use a timestamp of the tag's target commit: https://github.com/fluxcd/source-controller/pull/172/commits/ebfb1189b77b0a384d0ffea1f1717e84266a0129

**Comment by wingsofovnia** [comment, marked as answer]
https://github.com/fluxcd/source-controller/pull/172 Merged
